<!--  -->
<launch>

  <!-- ========= Bagplayer ========== -->
  <arg name="bagname" default="/home/razlaw/bags/obstacle_detection_dji/velodyne_dji_016-11-08-16-57-12_object_detection.bag"/>
  <arg name="rate" default="0.1"/>
  <arg name="start" default="0"/>

  <node pkg="rosbag" type="play" name="play" args="--clock -r $(arg rate) -s $(arg start) $(arg bagname)" output="screen" respawn="false" >
    <remap from="/velodyne_points" to="/velodyne_points_frombag" />
  </node>


  <!-- ========= Config Server ========== -->
  <!-- will try to read an initial configuration file form CONFIGPATH/config_ROBOTNAME -->
  <node name="config_server" pkg="config_server" type="config_server" output="screen" required="true">
    <param name="config_path" value="CONFIGPATH" />
    <param name="robot_name" value="ROBOTNAME" />
    <param name="robot_type" value="ROBOTTYPE" />
  </node>


  <!-- ============= Pointcloud Conversion (incl. nodelet_manager) =============  -->

  <!-- declare arguments with default values -->
  <arg name="pcap" default="" />
  <arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
  <arg name="min_range" default="0.4" />
  <arg name="max_range" default="130.0" />

  <node pkg="nodelet" type="nodelet" name="velodyne_nodelet_manager"
        args="manager" output="screen" />
  <arg name="read_once" default="false" />
  <arg name="read_fast" default="false" />
  <arg name="repeat_delay" default="0.0" />
  
  

  <arg name="frame_id" default="velodyne" />

  <node pkg="nodelet" type="nodelet" name="transform_nodelet"
        args="load velodyne_pointcloud/TransformNodelet velodyne_nodelet_manager"  output="screen">
    <param name="calibration" value="$(arg calibration)"/>
    <param name="min_range" value="$(arg min_range)"/>
    <param name="max_range" value="$(arg max_range)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
  </node>

  <!-- ============= Object Detection =============  -->
  <node pkg="nodelet" type="nodelet" name="velodyne_object_detector_nodelet" args="load velodyne_object_detector/velodyne_object_detector_nodelet $(arg frame_id)_nodelet_manager" output="screen" >
    <param name="points_topic" value="/velodyne_points"/>
    <param name="certainty_threshold_launch" value="0.0"/>
    <param name="median_small_kernel_size_launch" value="9"/>
    <param name="median_big_kernel_size_launch" value="41"/>
    <param name="distance_to_comparison_points_launch" value="3"/>

    <param name="publish_filtered_cloud" value="true"/>
  </node>


  <!-- ============= Rviz =============  -->
  <arg name="use_rqt" default="true"/>
  <node unless="$(arg use_rqt)" name="rivz" pkg="rviz" type="rviz" />
  <node if="$(arg use_rqt)" name="rqt_gui" pkg="rqt_gui" type="rqt_gui" output="screen" required="true" args=""/>
</launch>
